src/util/np_utils.py

```diff
@@ -0,0 +1,8 @@
+import numpy as np
+
+
+def normalized(v):
+    norm = np.linalg.norm(v)
+    if norm == 0:
+        norm = np.finfo(v.dtype).eps
```

Comment:
I worry that this behavior will make underlying software bugs unclear to a user who didn't happen to peak at the definition. I would prefer this allow the exception to occur.

Reply:
Why, since you should never be normalizing a zero vector? Is that considered bad behavior

Reply:
im not sure if it really makes sense to "normalize" a zero vec in the sense that it won't end up with unit length. If the user has some behavior that they want for the edge case they should check for it themself, but a generic function like normalize should except on the divide by zero.

Reply:
Ok yeah that makes sense

Reply:
Changed